packages/gatsby-plugin-mdx/utils/get-source-plugins-as-remark-plugins.js

```diff
@@ -20,66 +19,56 @@ module.exports = async function getSourcePluginsAsRemarkPlugins({
   ...helpers
 }) {
   debug(`getSourcePluginsAsRemarkPlugins`)
-  let pathPlugin = undefined
+
+  const userPlugins = []
+
   if (pathPrefix) {
-    pathPlugin = () =>
-      async function transformer(markdownAST) {
-        // Ensure relative links include `pathPrefix`
-        visit(markdownAST, `link`, node => {
-          if (
-            node.url &&
-            node.url.startsWith(`/`) &&
-            !node.url.startsWith(`//`)
-          ) {
-            // TODO: where does withPathPrefix
-            node.url = withPathPrefix(node.url, pathPrefix)
-          }
-        })
-        return markdownAST
-      }
+    userPlugins.push(async function withPathPrefixTransformer(markdownAST) {
+      // Ensure relative links include `pathPrefix`
+      visit(markdownAST, `link`, node => {
+        if (
+          node.url &&
+          node.url.startsWith(`/`) &&
+          !node.url.startsWith(`//`)
+        ) {
+          node.url = withPathPrefix(node.url, pathPrefix)
+        }
+      })
+      return markdownAST
+    })
   }
 
   fileNodes = getNodesByType(`File`)
 
   // return list of remarkPlugins
-  const userPlugins = gatsbyRemarkPlugins
-    .filter(plugin => {
-      if (_.isFunction(interopDefault(require(plugin.resolve)))) {
-        return true
-      } else {
-        debug(`userPlugins: filtering out`, plugin)
-        return false
-      }
-    })
-    .map(plugin => {
-      debug(`userPlugins: constructing remark plugin for `, plugin)
-      const requiredPlugin = interopDefault(require(plugin.resolve))
-      const wrappedPlugin = () =>
-        async function transformer(markdownAST) {
-          await requiredPlugin(
-            {
-              markdownAST,
-              markdownNode: mdxNode,
-              getNode,
-              getNodesByType,
-              files: fileNodes,
-              pathPrefix,
-              reporter,
-              cache,
-              ...helpers,
-            },
-            plugin.options || {}
-          )
+  gatsbyRemarkPlugins.forEach(plugin => {
+    const importedPlugin = interopDefault(require(plugin.resolve))
+    if (typeof importedPlugin !== `function`) {
+      debug(`userPlugins: not a valid plugin: \`${plugin}\``)
```

Comment:
Wouldn't this result in a "[object Object]" scenario? as plugin is an object?

Reply:
Hm yeah you're probably right :)